lib/widgets/message_list.dart

```diff
@@ -1105,9 +1110,9 @@ class StreamMessageRecipientHeader extends StatelessWidget {
         ]));
 
     return GestureDetector(
-      onTap: () => Navigator.push(context,
+      onTap: () => !isTopicNarrow? Navigator.push(context,
         MessageListPage.buildRoute(context: context,
-          narrow: TopicNarrow.ofMessage(message))),
+          narrow: TopicNarrow.ofMessage(message))): null,
```

Comment:
Pass `null` instead of `() => null` when we don't want to offer an on-tap interaction.

Also please see other examples of the [`condition ? expr1 : expr2` operator](https://dart.dev/language/operators#conditional-expressions) in the codebase—the spacing and indentation here doesn't match the project's style.